packages/cli/src/server/http-server.ts

```diff
@@ -17,7 +26,54 @@ export class HttpServer {
     });
   }
 
+  public redirectResponse(): Promise<string> {
+    return new Promise((resolve) => {
+      const interval = setInterval(() => {
+        if (this.token) {
+          clearInterval(interval);
+          resolve(this.token);
+        }
+      }, 300);
+    });
+  }
+
   public close(): void {
     this.server.close();
   }
+
+  private handleRedirectRequest(req: http.IncomingMessage) {
+    this.token = new URLSearchParams(req.url.slice(REDIRECT_ROUTE.length)).get('token');
+  }
+
+  private async handleWidgetDemo(res: http.ServerResponse): Promise<void> {
+    return new Promise((resolve) => {
+      const dashboardPath = path.resolve(__dirname, '../constants/dashboard/index.html');
+
+      fs.readFile(dashboardPath, 'utf8', (error, content) => {
+        if (error) {
+          if (error.code === 'ENOENT') {
+            res.end('404', 'utf-8');
+          } else {
+            res.end('500');
+          }
+        } else {
+          const payLoad = JSON.parse(this.config.getValue('triggerPayload'));
+
+          if (!payLoad) {
+            res.end('500');
```

Comment:
Would suggest doing a return here
```suggestion
            return res.end('500');
```

So the code execution won't continue

Reply:
Wow! you have a great eye, I added this one just now and missed the return command.